$NetBSD: patch-CMakeLists.txt,v 1.8 2026/10/11 06:41:44 adam Exp $

Do not append a buildlinked directory to ZIG_CMAKE_PREFIX_PATH to avoid WRKREF
in bin/zig.

Skip stage3 during build to:
* speed up building
* avoid work directory reference in zig's RUNPATH.

--- CMakeLists.txt.orig	2026-10-01 22:02:33.000000000 +0000
+++ CMakeLists.txt
@@ -18,13 +18,7 @@ endif()
 # cache variable. Furthermore, the environment variable follows the system path
 # separator, such as ':' on POSIX and ';' on Windows, but the cache variable
 # follows CMake's array behavior, i.e. always ';' for a separator.
-list(APPEND ZIG_CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH}")
-if(WIN32)
-  list(APPEND ZIG_CMAKE_PREFIX_PATH $ENV{CMAKE_PREFIX_PATH})
-else()
-  string(REGEX REPLACE ":" ";" ZIG_CMAKE_PREFIX_PATH_STRING "$ENV{CMAKE_PREFIX_PATH}")
-  list(APPEND ZIG_CMAKE_PREFIX_PATH "${ZIG_CMAKE_PREFIX_PATH_STRING}")
-endif()
+set(ZIG_CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH}")
 
 list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
@@ -790,18 +784,9 @@ if(NOT "${ZIG_TARGET_DYNAMIC_LINKER}" ST
 endif()
 
 
-add_custom_target(stage3 ALL
-  DEPENDS "${PROJECT_BINARY_DIR}/stage3/bin/zig"
-)
 
 set(ZIG2_WORKING_DIR "${PROJECT_SOURCE_DIR}")
 
-add_custom_command(
-  OUTPUT "${PROJECT_BINARY_DIR}/stage3/bin/zig"
-  COMMAND zig2 build ${ZIG_BUILD_ARGS} --prefix "${PROJECT_BINARY_DIR}/stage3"
-  COMMENT "Building stage3"
-  WORKING_DIRECTORY "${ZIG2_WORKING_DIR}"
-)
 
 set(ZIG_EXECUTABLE "$<TARGET_FILE:zig2>")
 
